models/workitem_repository.go

```diff
@@ -53,14 +57,14 @@ func (r *GormWorkItemRepository) Delete(ctx context.Context, ID string) error {
 		return NotFoundError{entity: "work item", ID: ID}
 	}
 	workItem.ID = id
-	tx := r.db
+	tx := r.db.Delete(workItem)
 
-	if err = tx.Delete(workItem).Error; err != nil {
-		if tx.RecordNotFound() {
-			return NotFoundError{entity: "work item", ID: ID}
-		}
+	if err = tx.Error; err != nil {
 		return InternalError{simpleError{err.Error()}}
 	}
+	if tx.RowsAffected == 0 {
```

Comment:
Not sure that would be correct. What if we have a collection-valued attribute in an object. Might that not lead to AffectedRows > 1?